mdata/store_mock.go

```diff
@@ -0,0 +1,48 @@
+package mdata
+
+import "github.com/raintank/metrictank/mdata/chunk"
+
+type mockSearchResult struct {
+	chunks []chunk.IterGen
+	err    error
+}
+
+type mockStore struct {
+	CurrCall int
+	Results  []mockSearchResult
+}
+
+func NewMockStore() *mockStore {
+	d := &mockStore{
+		CurrCall: 0,
+		Results:  make([]mockSearchResult, 0),
+	}
+	return d
+}
+
+func (c *mockStore) AddMockResult(chunks []chunk.IterGen, err error) {
+	// copy chunks because we don't want to modify the source
```

Comment:
right now i don't think there is a case where this happens. but if there is a case where somebody uses the `mockStore` and gets confused about the original values getting modified they would be glad about that.
at first i thought so too, that `copy` would only copy the slice and not the underlying values, but then i tested like this: https://play.golang.org/p/EpZk4uzObQ